spark/src/test/scala/geotrellis/spark/io/PersistenceSpec.scala

```diff
@@ -55,5 +68,21 @@ abstract class PersistenceSpec[K: ClassTag, V: ClassTag] extends FunSpec with Ma
       val expectedV: V = sample.filter(_._1 == key).values.first()
       readV should be equals expectedV
     }
+
+    // TODO: more specific exception type?
+    it("should delete a layer") {
+      deleter.delete(deleteLayerId)
+
+      // TODO: fix weird workaround, due to hadoop failure on count only
+      intercept[Exception] {
+        try {
+          reader.read(deleteLayerId)
```

Comment:
Hadoop should def throw at this point if the layer isn't found, must be a bug with the HadoopLayerReader